src/Administration/Resources/app/administration/src/core/service/entity-mapping.service.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * @sw-package framework
+ *
+ * @module core/service/validation
+ */
+
+interface PropertySchema {
+    type?: string;
+    entity?: string;
+    format?: string;
+    readOnly?: boolean;
+    properties?: Record<string, PropertySchema>;
+    [key: string]: unknown;
+}
+
+interface EntitySchema {
+    properties: Record<string, PropertySchema>;
+}
+
+type EntityNameMapping = Record<string, string>;
+type PropertyDefinition = Record<string, PropertySchema>;
```

Comment:
It would be good practice to export these types that are used in the public API of this service, e.g. as arguments / return types. Types that are only used internally in this file don't need to be exported.